srml/contract/src/lib.rs

```diff
@@ -498,6 +512,41 @@ decl_module! {
 			}
 		}
 
+		fn restore_to(origin, dest: T::AccountId, code_hash: CodeHash<T>, rent_allowance: BalanceOf<T>) {
+			let origin = ensure_signed(origin)?;
+
+			let origin_contract = <ContractInfoOf<T>>::get(&origin)
+				.and_then(|c| c.get_alive())
+				.ok_or("Cannot restore from inexisting or tombstone contract")?;
```

Comment:
This probably means that the tx was sent from EOA aka a non-contract account aka a regular user.

Reply:
we could extend the message if wanted